File: starter_kits/Go/src/hlt/Position.go
package hlt

import "fmt"

// Position - Location on the game map
type Position struct {
	x int
	y int
}

func NewPosition(x int, y int) *Position {
	return &Position{x, y}
}

func (p *Position) Coordinates() (int, int) {
	return p.x, p.y
}

func (p *Position) String() string {
	return fmt.Sprintf("Pos{x=%d,y=%d}", p.x, p.y)
}

// DirectionalOffset - Returns the position of a move in the direction
func (p *Position) DirectionalOffset(d *Direction) (*Position, error) {
	switch d.charValue {
	case NORTH:
		return &Position{p.x, p.y - 1}, nil
	case SOUTH:
		return &Position{p.x, p.y + 1}, nil
	case EAST:
		return &Position{p.x + 1, p.y}, nil
	case WEST:
		return &Position{p.x - 1, p.y}, nil
	case STILL:
		return &Position{p.x, p.y}, nil
	}
	return nil, fmt.Errorf("Invalid direction %c", d.charValue)
}

File: starter_kits/PHP/hlt/Position.php
<?php

namespace hlt;

class Position
{
    /**
     * @var  int
     */
    public $x;

    /**
     * @var  int
     */
    public $y;

    public function __construct(int $x, int $y)
    {
        $this->x = $x;
        $this->y = $y;
    }

    public function directionalOffset(Direction $d): Position
    {
        switch ($d) {
            case Direction::NORTH:
                $dx = 0;
                $dy = -1;
                break;
            case Direction::SOUTH:
                $dx = 0;
                $dy = 1;
                break;
            case Direction::EAST:
                $dx = 1;
                $dy = 0;
                break;
            case Direction::WEST:
                $dx = -1;
                $dy = 0;
                break;
            case Direction::STILL:
                $dx = 0;
                $dy = 0;
                break;
            default:
                throw new \RuntimeException('Unknown direction ' . $d);
        }

        return new Position($this->x + $dx, $this->y + $dy);
    }

    public function equals($other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }

        return $this->x === $other->x && $this->y === $other->y;
    }
}
